### ISSUE TITLE
Revamp attempt of the whole website

### ISSUE THREAD

**Issue by arvindravi** [libreprojects/libreprojects#104]
Hey there,

I'm trying to build the website from scratch with AngularJS, and i've removed all the jquery. Its been working out well so far, the basic structure is done. Features like favorites are yet to be implemented completely. The design, though, is the same, as I'm not very good with designing. 

If you're interested, feel free to contribute to the code [here](https://github.com/arvindravi/libreprojects). Thanks! :)

**Comment by jancborchardt** [comment]
What do the other @libreprojects/contributors think?

@arvindravi what are the advantages of this approach?

I think it’s cool that you want to contribute @arvindravi – however a framework like AngularJS seems to be overkill here. Something more lightweight like [Handlebars](http://handlebarsjs.com/) seems much more fitting, and will also be easier to learn for potential contributors. I heard from several programmers I respect that AngularJS is partly difficult to wrap your head around.
I would really prefer if we focus on the content rather than the underlying technology of the site, especially if it will be more difficult for contributors to get involved.

Sorry if this seems defensive – I’m really happy that you chose to contribute to Libre Projects! :) But we really need to weigh the pros and cons here.

**Comment by arvindravi** [comment]
Using AngularJS, makes the load times faster and eliminates the use of all the jquery plugins in there. Besides, I thought the implementation of _favorites_ could be much simpler and cleaner this way. Anyway, Its just a thought. 

Handlebars is again only a templating engine and needs some sort of javascript to be written, be it jquery or Angular or vanilla JS, and again I'm not denying its a overkill for what we have here, I just wanted to see what you guys think.

**Comment by jancborchardt** [comment]
@arvindravi ok, fair enough – if it’s faster then it seems a good change. I just want to make sure that the contributors are ok with it. I personally have just a tiny bit of experience (almost none) with AngularJS as of now.

@libreprojects/contributors what do you think?

**Comment by airon90** [comment]
I don't have the knowledge to evaluate the proposal. I read the latest comment of @arvindravi and I trust him. +1

If there will be some problem we will rollback to latest useful version of the code.

**Comment by ironbishop** [comment]
I don't know AngularJS, but I can learn. This could be a good time to improve mobile layout, fix #70, an maybe work on #16.

**Comment by arvindravi** [comment]
@ironbishop FWIW, #70 is already fixed in my code. I was going to complete the whole thing and post here, but I couldn't find enough time to work on it, and I thought it'd be nice if I had some help. The favourites part and some css parts of the modal are the things left to be worked on. So, yeah, you guys decide if you want to take it up. And we can take this further.

**Comment by jancborchardt** [comment]
It would be nicer to have some smaller pull requests instead of trying to change everything at once. :)

**Comment by yuwash** [comment]
I don’t have any knowledge about AngularJS either but the code looks fairly clean. Maybe could we publish it simultaneously as beta to test out a bit? 
How do you suggest to break up this change involving a change of the framework?

**Comment by arvindravi** [comment]
Migrating this change is one thing I'm still confused about, because the work is not as complete as the one thats live right now. What I have in mind is, maybe push this to `http://libreprojects.net/test/` or something and work on it until its complete?

And because the entire framework is going to be changed, I'm not sure how smaller pull requests can be issued here. You guys have any thoughts?

Also, please think this through, as the contributors might need to get comfortable with AngularJS, its fairly easy though.

**Comment by jancborchardt** [comment]
Ok, since it’s a framework change it makes sense to make it all at once.

And yes, pushing to libreprojects.net/new would be good. We should not replace the actual site until everyone of us is happy with it, and AngularJS.

**Comment by airon90** [comment]
1. Is the new version already online somewhere?
2. @arvindravi: Does the migration to AngularJS change code syntax in the management of the projects?

**Comment by arvindravi** [comment]
@Airon90 If you're referring to, how to update/remove projects, I haven't changed that, we can still manage the projects by updating `data.json`.

**Comment by airon90** [comment]
Yes, I was referring to that. This is my only interest on this question, because I'm not a developer :(

**Comment by jancborchardt** [comment]
@arvindravi after the change is done, you probably need to maintain the site for some time until all contributors are up to speed. Also, good code documentation is needed so others can work on it as well. :)

**Comment by arvindravi** [comment]
@jancborchardt Like I said already, the favorites parts is still incomplete. I currently don't have the time to work on it. So, I believe maybe when that's complete (by me or by someone else), I can make sure that I take care of it until others are up to speed.
